[Ong Wei Jie] ip

ongweijie7 opened the pull request:
DukePro frees your mind of having to remember things you need to do. It's,

- ~~Fast~~ *SUPER FAST* to use for CLI users
- User-Friendly
-  Allows you to keep track of your schedule!!

ALL YOU NEED TO DO IS TO download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).

1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is **FREE**!

Features:
 - [x] Managing tasks
 - [x] Finding Tasks
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ongweijie7 committed:
Added greet, echo and exit

ongweijie7 committed:
no message

ongweijie7 committed:
Added the ability to store text entered by the user and display when requested

ongweijie7 committed:
Added ability to mark list as done/not done

ongweijie7 committed:
Added different subclasses of Task

ongweijie7 committed:
Added ToDos, Deadlines, Events classes

ongweijie7 committed:
Added exceptions when user input is incorrect

ongweijie7 committed:
Added capability to delete in list

ongweijie7 committed:
Tidy up code

Replaced some if-else statements with case.

Replaced remove() method in task class and subclasses to removeTask().

ongweijie7 committed:
Add ability to save tasks in hard disk

ongweijie7 committed:
Add ability for duke to process dates

ongweijie7 committed:
Merge branch 'branch-level-7'

* branch-level-7:
  Add ability to save tasks in hard disk

# Conflicts:
#	src/main/java/Duke.java

ongweijie7 committed:
Add more classes for OOP

ongweijie7 committed:
Implement A-MoreOOP

ongweijie7 committed:
Organise into packages

ongweijie7 committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

* commit '556af3f47a96b32898ab4cdbd65b16486a4871e8':
  Add Gradle support

ongweijie7 committed:
Add Junit tests and JAR file

ongweijie7 committed:
no message

ongweijie7 committed:
Add javadoc and tweak code to comply with coding standard

ongweijie7 committed:
Add command to find keyword

afiqzu commented on src/main/java/Duke.java:
```diff
@@ -50,18 +54,29 @@ else if (input.equals("list")) {
+                            input = taskDate[0] + "(by:" + dateTimeString  + ")";
                             newTask = new Deadline(input);
                             break;
                         case "event" :
```
Just a minor coding standard violation but I think there should not be a white space between the "event" condition and the colon

afiqzu commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,39 @@
+                        break;
+                    case "D" :
+                        this.actions.add(new Deadline(taskName));
+                        break;
```
Perhaps you would like to insert a default case to make it clear that it is the end of the switch statement?

afiqzu reviewed (commented):
Overall your code is easy to understand 👍 Keep it up!

zhoyx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     * @throws Missing if there is missing information regarding date and time
+     * @throws WrongKeyWord if there is a missing key command
+     */
+    public static String understandInput(String input, String[] input_arr, TaskList tasks, Storage storage) throws Missing, WrongKeyWord {
```
Would both input and input_arr both need to be passed in to the function? Could you obtain one from the other?

zhoyx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+public class Parser {
+
+    /**
+     * Returns reply string based on.
```
Is the fullstop here intended?

zhoyx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+     * @param filePath path of where the file is.
+     */
+    public Storage(String filePath) {
+        this.path = "./src/main/data/duke.txt";
```
Should this make use of the filePath passed into the constructor 🤔

zhoyx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+ * Class that prints output for the user.
+ */
+public class Ui {
+    private Scanner r;
```
Maybe a more descriptive name for the scanner would make it clearer.

zhoyx commented on src/main/java/duke/dukeexceptions/TaskListEmpty.java:
```diff
@@ -0,0 +1,26 @@
+package duke.dukeexceptions;
+
+/**
+ *  An exception that throws when TaskList is empty.
+ */
+public class TaskListEmpty extends DukeException{
+
+    /**
+     * Constructor for the Exception.
+     *
+     * @param s string to be thrown during exception.
+     */
+    public TaskListEmpty(String s) {
+        super(s);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return a warning message when there is missing information.
+     */
+    @Override
+    public String getMessage() {
+        return "☹ hmm strange the list is missing and so are my cookies?!?";
+    }
+}
```
Good use of various duke exceptions for error handling 👍

zhoyx commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,49 @@
+     */
+    public String removeTask() {
+        total--;
+        return "Noted I've removed this task:\n "
+                + this.status() + "\n" + "Now you have "
+                + super.total + " tasks in the list";
```
Clean line breaks, improves readability 👍

zhoyx commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,49 @@
+        return type + super.status();
+    }
+
+    /**
+     * Returns the updates on removal
+     *
+     * @return string regarding removal of a ToDo
+     */
```
Maybe consider adding a fill stop to the end of JavaDocs sentences

zhoyx reviewed (commented):
Overall, I fould your code easy to read. Just ensure JavaDocs end off with fullstops.

ongweijie7 committed:
Add GUI interface

ongweijie7 committed:
Correct coding violations and use CheckStyle

ongweijie7 committed:
Add assertion and created .jar file

ongweijie7 committed:
Add Assertions and also catch statements to catch various exceptions

ongweijie7 committed:
Improve code quality

There was a complicated "if else" expression to determind the types of task as there were a few or expressions.

Breaking down the "if else" expression into their own individual expressions improved the readability and code quality.

ongweijie7 committed:
Improve code quality and readability in Parser

There was a complicated "if else" expression to determind the types of task as there were a few or expressions to check for all types of task.

Breaking down the "if else" expression into their own individual expressions improved the readability and code quality.

This ensures that there are seperate conditional expressions to consider each variation of task added, allowing for easier understanding of code.

ongweijie7 committed:
Merge pull request #2 from ongweijie7/branch-A-Assertions

Branch a assertions

ongweijie7 committed:
Revert "Branch a assertions"

ongweijie7 committed:
Merge pull request #3 from ongweijie7/revert-2-branch-A-Assertions

Revert "Branch a assertions"

ongweijie7 committed:
Merge branch 'master' of https://github.com/ongweijie7/ip

* 'master' of https://github.com/ongweijie7/ip:
  Revert "Branch a assertions"
  Add Assertions and also catch statements to catch various exceptions
  Add assertion and created .jar file

ongweijie7 committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

* origin/master:
  Revert "Branch a assertions"
  Add Assertions and also catch statements to catch various exceptions
  Add assertion and created .jar file

ongweijie7 committed:
Merge remote-tracking branch 'origin/branch-A-CodeQuality'

* origin/branch-A-CodeQuality:
  Improve code quality

ongweijie7 committed:
Add a few catch statements in Parser and edited the error message thrown.

Added catch statements to catch exceptions because they could be thrown when user gave missing commands.

Exceptions that occur when there is a missing time/date or wrong format given will be caught and I will throw a Missing DukeException.

It is done this way since this error is specific to a missing command.

The error message for Missing exception is also improved as I realised the missing command is also used when there is no information on the task name given

ongweijie7 committed:
Add a few catch statements in Parser and edited the error message thrown.

Added catch statements to catch exceptions because they could be thrown when user gave missing commands.

Exceptions that occur when there is a missing time/date or wrong format given will be caught and I will throw a Missing DukeException.

It is done this way since this error is specific to a missing command.

The error message for Missing exception is also improved as I realised the missing command is also used when there is no information on the task name given

ongweijie7 committed:
Remove 3 empty lines

ongweijie7 committed:
Remove 3 empty lines

ongweijie7 committed:
Merge commit '306bf4fd7b532958292332396c6a4e359c29ac73'

* commit '306bf4fd7b532958292332396c6a4e359c29ac73':
  Remove 3 empty lines
  Add a few catch statements in Parser and edited the error message thrown.
  Improve code quality

ongweijie7 committed:
Make changes to Parser, TaskList and Storage classes.

Abstracted away the marking, unmarking and deleting of tasks. Also make it such that storage will keep track of done tasks.

TaskList should handle the marking of tasks instead of letting tasks handle it themselves.

It is done this way because Storage needs to constantly update the textfile.

ongweijie7 committed:
Add syntax to make commands simpler.

Needs to be changed to make things easier for user especially for command line users

Have shortened the first command to a simple capital letter. This ensures that actual commands can be differentiated from accidental inputs.

ongweijie7 committed:
Make minor deletes to Bye commands

ongweijie7 committed:
Add the user guide for Duke

ongweijie7 committed:
no message

ongweijie7 committed:
no message

ongweijie7 committed:
Make improvements to storage

ongweijie7 committed:
no message

ongweijie7 committed:
Change the User guide

ongweijie7 committed:
no message

ongweijie7 committed:
Edit UserGuide

ongweijie7 committed:
Make minor improvements

ongweijie7 committed:
no message
